core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -36,25 +36,51 @@ class Replica(val brokerId: Int,
   // for local replica it is the log's end offset, for remote replicas its value is only updated by follower fetch
   @volatile private[this] var logEndOffsetMetadata: LogOffsetMetadata = LogOffsetMetadata.UnknownOffsetMetadata
 
+  // The log end offset value at the time the leader receives last FetchRequest from this follower.
+  // This is used to determine the last catch-up time of the follower
+  @volatile private var lastFetchLeaderLogEndOffset: Long = 0L
```

Comment:
I think it's better to initialize the replica to the notCaughtUp state, i.e., lastCaughtUpTime = 0, lastFetchLeaderLogEndOffset = MAX_LONG and lastFetchTimeMs = 0?